src/ooxml/java/org/apache/poi/xwpf/usermodel/XWPFDocument.java

```diff
@@ -1332,7 +1346,7 @@ public void setZoomPercent(long zoomPercent) {
     public void insertTable(int pos, XWPFTable table) {
         bodyElements.add(pos, table);
         int i = 0;
-        for (CTTbl tbl : ctDocument.getBody().getTblArray()) {
```

Comment:
despite being deprecated, the array methods are better than the list methods - we have a job that cherry picks the necessary CT generated classes and it works better with methods that return concrete results like CTTable[] but misses List<CTTable> which is backed by an inner class (the one our job misses)